### ISSUE TITLE
SSsummarize() not working for SS ver3.30

### ISSUE THREAD

**Issue by John-R-Wallace** [r4ss/r4ss#153]
SSsummarize in r4ss ver 1.24.0 on CRAN works for SS 3.24 but not 3.30:

The latest r4ss on GitHub also didn't work.

> SSsummarize(SSgetoutput(dirvec = c("W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/0-DBRK 2015 Base"), underscore=T))
length(dirvec) as input to SSgetoutput: 1 
reading output from W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/0-DBRK 2015 Base/Report.sso 
added element 'replist1' to list
imodel=1/1,  N active pars=203

> SSsummarize(SSgetoutput(dirvec = c("W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/1-DBRK 2015 Base SS3.30"), underscore=T))
length(dirvec) as input to SSgetoutput: 1 
reading output from W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/1-DBRK 2015 Base SS3.30/Report.sso 
Error in read.table(wtfile, header = TRUE, comment.char = "", skip = (grep("yr seas gender",  : 
  duplicate 'row.names' are not allowed
In addition: There were 11 warnings (use warnings() to see them)

**Comment by CWetzel** [comment]
Generally, this function is working for 3.30.  However, it was not working
(as of a couple weeks ago) to summarize models that have different start
and end years.  Do the models that you are attempting to summarize fall
into this category?  Ian T. had been alerted to this issue.

**Comment by John-R-Wallace** [comment]
All have the same years except the 2015 base changed to SS ver 3.30.

**Comment by John-R-Wallace** [comment]
Vlada must have an older version, it's been working for her.

**Comment by CWetzel** [comment]
I just downloaded the latest r4ss version.  It works on models with the
same and different model years.  Thank you, Ian for fixing that issue.

John - I don't have an answer for why it is not working on your models.

**Comment by taylori** [comment]
The error is in SS_output (which is called by SSgetoutput), while trying to read wtatage.ss_new. Either something is strange in that file, or it's format has changed in whatever version of SS is being used. If there's no something obviously wrong with the wtatage.ss_new file, could you send me the model input files along with Report.sso and wtatage.ss_new?

**Comment by taylori** [comment]
Also, assuming that the error is only with reading wtatage.ss_new, which contains information that you probably don't need, you can skip it via
`SS_output(..., readwt=FALSE)`

Note that the SSgetoutput function is just a wrapper for SS_output when you have lots of models to read. For fewer models, it's probably best to run SS_output anyway. The following lines would be equivalent to pointing SSgetoutput to dirvec = c(dir1, dir2, dir3), but allows you to add a 4th model without re-reading the first 3.

```
mod1 <- SS_output(dir1)
mod2 <- SS_output(dir2)
mod3 <- SS_output(dir3)

mod.summary <- SSsummarize(list(mod1, mod2, mod3))
```

**Comment by John-R-Wallace** [comment]
|SS_output() is working fine without using '||readwt=FALSE'.

||It was late yesterday and I only sent the error from the r4ss on 
CRAN.  Below is the error using the latest version of r4ss on GitHub, 
with the first model reading in being SS ver 3.24 and the second model 
version 3.30.
|
|length(dirvec) as input to SSgetoutput: 2
reading output from 
W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/0-DBRK 2015 
Base/Report.sso
added element 'replist1' to list
reading output from 
W:/ALL_USR/JRW/Assessment/Darkblotched/2017/Models/10-DBRK 2017 
Catch,Steepness, DRate,Dcomps, CBO VAST Final comps, BDScomps, 
AShpLAcomps/Report.sso
! Warning: no catch-at-age numbers because 'detailed age-structured reports'
            is turned off in starter file.
added element 'replist2' to list
imodel=1/2Error in `$<-.data.frame`(`*tmp*`, "imodel", value = 1L) :
   replacement has 1 row, data has 0
In addition: Warning messages:
1: NAs introduced by coercion
2: NAs introduced by coercion
3: NAs introduced by coercion
|
|# ------------------------------------------------------------------------
|
|The ||SSsummarize(list(mod1, mod2, mod3)) way does work for me and 
below is an automated version, |||with legend labels, |using Model.1, 
Model.2, ... :|
|
modNums <- c(1, 4:10)

Model.List <- list()
for (i in 1:length(modNums))
       Model.List[[paste0("Model.",modNums[i])]] <- eval(parse(text = 
paste0("Model.",modNums[i])))

SSplotComparisons(SSsummarize(Model.List), subplot=1:10, legendlabels = 
paste0("Model.", modNums))
||
|
|Here is the same all put together:|

|modNums <- c(1, 4:10)|
|SSplotComparisons(SSsummarize({Model.List <- list(); for (i in 
1:length(modNums)) Model.List[[paste0("Model.",modNums[i])]] <- 
eval(parse(text = paste0("Model.",modNums[i]))); Model.List}),
                          subplot=1:10, legendlabels = paste0("Model.", 
modNums))
|||
||
||

The SSgetoutput()  method (currently not working for me), all put 
together is:|
|
modNums <- c(1, 4:10)
SSplotComparisons(SSsummarize(SSgetoutput(dirvec = {Models <- NULL; for( 
i in modNums ) Models <- c(Models, eval(parse(text = paste0("M",i)))); 
Models}, underscore=T)), subplot=1:10, legendlabels = paste0("Model.", 
modNums))

# 
----------------------------------------------------------------------------------------------------------

Ian, I will call and if needed, send you the files.

Thanks,

-John

**Comment by taylori** [comment]
@John-R-Wallace, thanks for the more detailed information. 

I was able to produce the same error in SSgetoutput, which seems to be the problem, but it's a bit mysterious why the models read that way are missing information. Also, it's not specific to 3.24 or 3.30. I won't be able to work on it more today but can get to it soon. Glad that the work-around works for now.

**Comment by taylori** [comment]
Now fixed in commit a131de5. Difference between SSgetoutput and multiple calls to SS_output was the default for SSgetoutput was forecast=FALSE, which was trimming some of the output tables using an out-of-date column name. I've fixed the column name mismatch (was related to issue #141), but also aligned the defaults to both be forecast=TRUE (in commit c54c9ea).

**Comment by John-R-Wallace** [comment]
Thanks Ian!

**Comment by taylori** [comment]
Fix to this issue was only applied to size-based selectivity. Thanks to @ciallen for reporting the issue.

**Comment by taylori** [comment]
Should now be fixed in commit b8640f909668d6762adadbccc1619cb68cdea8fa.
